### DISCUSSION TITLE
Quality control in R

### DISCUSSION THREAD

**Discussion by sanaashrafali** [now-i-know-my-abcd/docs#25] [Stats & Code Help]
Hi everyone! I am trying to run some quality control in R, and I've hit a wall. I'm trying to recode multiple columns of ROI beta weights to NA on the condition that they don't meet QC standards. Does anyone have code for this? I've started by trying with just one column, but to no avail. This is what I have so far:

<img width="449" alt="Screen Shot 2022-08-03 at 10 16 48 AM" src="https://user-images.githubusercontent.com/103009041/182669488-1f4e51d8-c9b3-40ef-9fca-09c0be46873c.png">

<img width="317" alt="Screen Shot 2022-08-03 at 10 17 16 AM" src="https://user-images.githubusercontent.com/103009041/182669580-7e66c301-6290-4054-8c71-d0e13d7e5d90.png">

Follow up question—can I make that work for multiple columns at a time, maybe by creating a vector of column names/numbers it applies to?

<img width="532" alt="Screen Shot 2022-08-03 at 10 18 41 AM" src="https://user-images.githubusercontent.com/103009041/182669867-51882015-69b6-4b3e-baee-7f0d0a38b075.png">

**Comment by jflournoy** [comment]
Hi @sanaashrafali,

One thing that can help with code questions is if you can put the code in blocks between 
```` 
```
Three backticks
```
````

That makes it easy top copy and paste and play with.

This looks like a good answer to your question: https://stackoverflow.com/questions/39209987/using-functions-of-multiple-columns-in-a-dplyr-mutate-at-call

One issue in the code above that may have made this extra troublesome is the conditional checks on `mrif_score`. In

```r
case_when(
      fsqc_qc != 1 ~ NA_real_,
      mrif_score != 1 ~ NA_real_,
      mrif_score != 2 ~ NA_real_,
      iqc_rsfmri_ok_ser < 0 ~ NA_real_,
      TRUE ~ var)
```

you first check if mrif_score does not equal one, and if so, assign an NA (which stops evaluation of the other conditions in case_when). If it _is_ equal to one, it goes to the next check where if mrif_score is not equal to 2 it gets an NA. If it passes the first check, we know it is equal to one, so it can't be equal to 2. So essentially this gives an NA for any value of mrif_score (no matter what the number is, it can't be equal to both 1 and 2). 

I think that might have been the _only_ trouble with your solution but I've put a reproducible example below just in case.

```r
library(tidyverse)

set.seed(101)
d <- tibble(fsqc_qc = rbinom(100, 1, prob = .5),
            mrif_score = rbinom(100, 2, prob = .5),
            iqc_rsfmri_ok_ser = -rbinom(100, 1, prob = .1),
            bold1 = rnorm(100),
            bold2 = rnorm(100),
            bold3 = rnorm(100),
            bold4 = rnorm(100))

d %>% 
  mutate(
    across(c(bold1, bold2, bold3, bold4),
    ~case_when(
      fsqc_qc != 1 ~ NA_real_,
      ! mrif_score %in% c(1, 2) ~ NA_real_,
      iqc_rsfmri_ok_ser < 0 ~ NA_real_,
      TRUE ~ .
    )))
```

which gives:

```
# A tibble: 100 × 7
   fsqc_qc mrif_score iqc_rsfmri_ok_ser  bold1  bold2  bold3  bold4
     <int>      <int>             <int>  <dbl>  <dbl>  <dbl>  <dbl>
 1       0          0                 0 NA     NA     NA     NA    
 2       0          0                 0 NA     NA     NA     NA    
 3       1          1                 0 -0.753  0.820 -1.11  -1.73 
 4       1          2                 0  0.732 -0.345  1.06   0.263
 5       0          1                -1 NA     NA     NA     NA    
 6       0          1                 0 NA     NA     NA     NA    
 7       1          0                 0 NA     NA     NA     NA    
 8       0          0                 0 NA     NA     NA     NA    
 9       1          2                 0 -0.270 -1.04   0.773 -0.470
10       1          2                 0  0.249 -0.147 -1.41   0.124
# … with 90 more rows
# ℹ Use `print(n = ...)` to see more rows
```

**Comment by landryhuffman** [comment]
Hey Sana! Here's an example of my QC code. With release 4.0, they combined all QC measures into one binary index, so it's a little more simple (although before that I did QC as John recommends it). 

`lts_roi.1 <- lts_roi.1 %>%
  mutate(across(
    c("wm_rmf_l", "wm_rmf_r", "wm_ip_l", "wm_ip_r", "wm_rac_l", "wm_rac_r", 
      "wm_cac_l", "wm_cac_r", "wm_lo_l", "wm_lo_r", "wm_rmf_l_n", "wm_rmf_r_n", 
      "wm_rac_l_n", "wm_rac_r_n", "wm_cac_l_n", "wm_cac_r_n", "wm_rmf_m", 
      "wm_rac_m", "wm_cac_m", "wm_ip_m", "wm_lo_m", "emo_amy_l", "emo_amy_r", 
      "emo_rac_l", "emo_rac_r", "emo_cac_l", 
      "emo_cac_r", "emo_ff_l", "emo_ff_r", "emo_ins_l", "emo_ins_r", 
      "emo_amy_l_n", "emo_amy_r_n", "emo_cac_l_n", "emo_rac_m", "emo_ff_m", 
      "emo_ins_m", "emo_amy_m", "emo_cac_m"),
    ~case_when(
      imgincl_nback_include != 1 ~ NA_real_,
      TRUE ~ .)
  ))
`

**Comment by katedamme** [comment]
Does anyone have a sense of what the notes codes stand for (e.g. sus1, mo2)? I am trying to get a sense of why some of these images are excluded.
